File: src/types/equipment/ArmorType.ts
/**
 * @main ArmorType
 */

import { validateSchemaCreator } from "../../validation/schema.js"
import { LocaleMap } from "../_LocaleMap.js"
import { NonEmptyString } from "../_NonEmptyString.js"

/**
 * @title Armor Type
 */
export type ArmorType = {
  /**
   * The armor type's identifier. An unique, increasing integer.
   * @integer
   * @minimum 1
   */
  id: number

  /**
   * Each armor type has a *sturdiness rating*. The higher the rating, the more
   * durable the armor. Rolling higher than this rating during a sturdiness
   * check means the armor receives one level of the new condition *Wear*.
   * @integer
   * @minimum 1
   * @msximum 20
   */
  sturdiness_rating: number

  /**
   * All translations for the entry, identified by IETF language tag (BCP47).
   */
  translations: LocaleMap<ArmorTypeTranslation>
}

export type ArmorTypeTranslation = {
  /**
   * The name of the armor type.
   */
  name: NonEmptyString
}

export const validateSchema = validateSchemaCreator<ArmorType>(import.meta.url)

File: Sources/OptolithDatabaseSchema/Types/equipment/item/sub/ArmorType.swift
import FileDB

@Model
public struct ArmorType {
  /// An armor type can have a *sturdiness rating*. The higher the rating, the more durable the armor. Rolling higher than this rating during a sturdiness check means the armor receives one level of the new condition *Wear*.
  let sturdiness_rating: Integer({ minimum: 1, maximum: 20 })?

    /// All translations for the entry, identified by IETF language tag (BCP47).
    @Relationship
    let translations: [String: Translation]

    struct Translation { // ArmorTypeTranslation

        /// The armor type’s name.
        let name: String({ minLength: 1 })
    }
}
